kbfscrypto/crypto_key_types.go

```diff
@@ -644,3 +669,33 @@ func UnmaskBlockCryptKey(serverHalf BlockCryptKeyServerHalf,
 	tlfCryptKey TLFCryptKey) BlockCryptKey {
 	return MakeBlockCryptKey(xorKeys(serverHalf.data, tlfCryptKey.data))
 }
+
+// BlockHashKey is used as input to encrypt/decrypt block data (v2).
+type BlockHashKey struct {
+	// Should only be used by implementations of Crypto.
+	privateByte64Container
+}
+
+// MakeBlockHashKey makes a key used for encryption and decryption for
+// the v2 block encryption scheme.
+func MakeBlockHashKey(
+	serverHalf BlockCryptKeyServerHalf, key TLFCryptKey) BlockHashKey {
+	keyData := key.Data()
+	mac := hmac.New(sha512.New, keyData[:])
```

Comment:
this whole block can be done slightly simpler, I think:

```
hash64 := sha512.Sum512(append(key.Data(), serverHalf.Data()...))
```

Reply:
I could be wrong, but I think what you propose is pretty different from doing it via HMAC, right?  Max explicitly requested doing HMAC-SHA512 for this.

Reply:
Ah, you're right! never mind.